### ISSUE TITLE
PythonSlicer problem with vtk and ReadData

### ISSUE THREAD

**Issue by fbordignon** [Slicer/Slicer#6484]
The code block below reports an error when run with PythonSlicer.exe on Windows 10
```python
import slicer,mrml
A = slicer.vtkMRMLNRRDStorageNode()
A.SetFileName("foo")
n = mrml.vtkMRMLScalarVolumeNode()
A.ReadData(n)
```
`TypeError: ReadData argument 1: method requires a VTK object`

This occurred on Slicer 5 nightly build 31096

I suspect this issue is also affecting SlicerCAT built custom apps CLIs, since my custom app clis are not being able to ReadData() as well.

The code block above used to work on Slicer 4.

Another issue:
```python
import mrml
mrml.vtkMRMLScalarVolumeNode().GetClassName()
```
`AttributeError: 'MRMLCore.vtkMRMLScalarVolumeNode' object has no attribute 'GetClassName'`

**Comment by jamesobutler** [comment]
Using Slicer 5.1.0-2022-07-26 (Revision 30196) and also latest Slicer stable 5.0.3 (Revision 30893) I'm not able to replicate these issues. I downloaded both of these versions from https://download.slicer.org/.

Notably this was on Windows 11, but I don't see why Windows 10 would cause this issue for you.

![image](https://user-images.githubusercontent.com/15837524/181579094-6936878b-b9a9-48d7-8009-312d1cbd33ac.png)

**Comment by jamesobutler** [comment]
https://discourse.slicer.org/t/how-to-import-slicer-in-python-cli/8083/3?u=jamesobutler

> Generally, CLIs are not intended to have direct access to the Slicer scene / infrastructure. They are meant for non-interactive data processing with non-slicer libraries. If you want to work within the Slicer environment, you should keep the code within a python scripted module.

**Comment by fbordignon** [comment]
I am sorry if I was not very clear the first time. The problem occurs when running this comandos using PythonSlicer.exe console ran independently from slicer.

**Comment by pieper** [comment]
Importing the python-wrapped VTK C++ code can be tricky - if they aren't imported in the correct order then the methods may not be available and sometimes you need to import the libraries directly.  This probably hasn't been looked at before for the PythonSlicer environment with the new VTK.  I'm guessing it is solvable with a script to manage the import process or some change in the way they are packaged.

**Comment by fbordignon** [comment]
Thanks, @pieper your comment lead me to try different import orders and it seems that it works if I import vtk first and everything in the same line:

```python
import vtk, slicer, slicer.util, mrml
A = slicer.vtkMRMLNRRDStorageNode()
A.SetFileName("foo")
n = mrml.vtkMRMLScalarVolumeNode()
A.ReadData(n)
```
The error now is relative to the incorrect filename "foo" but if I select a valid file, everything seems to work and ReadData returns 1. GetClassName also works

**Comment by lassoan** [comment]
My understanding is that the very first time you instantiate a Python-wrapped VTK class, the wrapper traverses up the class hierarchy to determine the base classes of the object. This is done only once, at the first instantiation. If at that point you haven't imported all the base classes imported then you'll never have full hierarchy information for that instantiated class.

Therefore, you need to import vtk before you instantiate VTK-based classes.

I've checked and calling `import vtk` before your code snippet fixes the `TypeError: ReadData argument 1: method requires a VTK object` error.

**Comment by fbordignon** [comment]
Thanks @lassoan, it is indeed the import order like I've described above. Since it was working on Slicer 4 without importing vtk, I was thinking it could be a bug. Anyways, this could serve as a reference for people that bumps into the same issue.
Regards, Fernando.

**Comment by lassoan** [comment]
Since many parts of Slicer (all module logics, PythonQt, etc.) are loaded by the Slicer application, currently it is quite limited what can be done in a Python environment that is not created by the Slicer application. That's why we don't put too much effort into documenting this.

But in the long term we want to make more and more parts of Slicer usable without the Slicer application (starting with MRML, DICOM, module logic classes). So, it is good that you explore this and we would be open to making improvements that helps moving in this direction.
